src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -649,6 +649,12 @@ private void ParseSpecs(string? metricsSpecs)
                     return;
                 }
 
+                if (metricsSpecs.Length == 0)
+                {
+                    _aggregationManager!.IncludeAll();
+                    return;
+                }
```

Comment:
Could this be enabling all metrics in a process that was otherwise subscribed to none?

I added a simple
```c#
var meter = new Meter("TestMeter");
var testCounter = meter.CreateCounter<int>("test.testCounter");

_ = Task.Run(async () =>
{
    while (true)
    {
        await Task.Delay(1000);
        Console.WriteLine($"Enabled: {testCounter.Enabled}");
    }
});
```

into the benchmark project in question and it appears to be enabled after this change.

Reply:
Previously my understanding was that one of Blanch's changes during Preview7 had regressed this behavior and @tarekgh's change here was intended to restore the previous .NET 8 behavior. However doing some testing now, thats not what I see. What I see in my test for .NET 8 is omitting the Metrics argument did not enable any metrics.

We probably need to dig deeper and make sure we understand exactly what happened, but now it looks like this change that was intended to re-align us to .NET 8 behavior instead changed us away from .NET 8 behavior. This is my test case:

```C#
internal class Program
{
    static void Main(string[] args)
    {
        Meter m = new Meter("MyMeter");
        m.CreateObservableCounter("MyCounter", () => 1);
        MyEventListener listener = new MyEventListener();
        Console.ReadLine();
    }
}

class MyEventListener : EventListener
{
    protected override void OnEventSourceCreated(EventSource eventSource)
    {
        Console.WriteLine(eventSource.Name);
        if(eventSource.Name == "System.Diagnostics.Metrics")
        {
            EnableEvents(eventSource, EventLevel.Informational, (EventKeywords)0x7/*, new Dictionary<string,string?>() { { "Metrics", "MyMeter" } }*/);
        }
    }

    protected override void OnEventWritten(EventWrittenEventArgs eventData)
    {
        if(eventData.EventSource.Name == "System.Runtime")
        {
            return;
        }
        Console.WriteLine(eventData.EventName);
        for (int i = 0; i < eventData.Payload.Count; i++)
        {
            Console.WriteLine($"\t{eventData.PayloadNames[i]}: {eventData.Payload[i]}");
        }
    }
}
```

Reply:
I tried this code on .NET 8.0 and I am seeing `MyMeter` get reported as expected:

```
Microsoft Windows 10.0.22631
.NET 8.0.8
System.Diagnostics.DiagnosticSource.dll 8.00.824.36612
```

```
Microsoft-Windows-DotNETRuntime
System.Runtime
System.Buffers.ArrayPoolEventSource
System.Diagnostics.Metrics
Message
        Message: New session started. SessionId auto-generated: 919cfab8-dc80-4eb5-af66-0ba09cfb9e42
Message
        Message: No RefreshInterval argument received. Using default 1s.
Message
        Message: No MaxTimeSeries argument received. Using default 1000
Message
        Message: No MaxHistograms argument received. Using default 20
Message
        Message: No Metrics argument received
InstrumentPublished
        sessionId: 919cfab8-dc80-4eb5-af66-0ba09cfb9e42
        meterName: MyMeter
        meterVersion:
        instrumentName: MyCounter
        instrumentType: ObservableCounter`1
        unit:
        description:
        instrumentTags:
        meterTags:
        meterScopeHash:
```

Reply:
You will always see the InstrumentPublished events for all metrics, but you won't see the CounterRatePayload events that transmit the values of the counter. Sending CounterRatePayload is what requires the AggregationManager to allocate storage.

Reply:
I have submitted the PR https://github.com/dotnet/runtime/pull/107267 to fix the issue.